module/zfs/dsl_pool.c

```diff
@@ -98,8 +98,17 @@
  * capped at zfs_dirty_data_max_max.  It can also be overridden with a module
  * parameter.
  */
+/* Module Params */
 unsigned long zfs_dirty_data_max = 0;
 unsigned long zfs_dirty_data_max_max = 0;
+
+/*
+ * Values computed based on total memory unless
+ * the above module parameters are set
+ */
+unsigned long dirty_data_max = 0;
```

Comment:
with this change in design, I don't see how we can correct a bad setting or inappropriate limit (eg 4GB above) on a live system without recompiling. In other words, I see the change of zfs_dirty_data_max[_max] to dirty_data_max as gratuitous. Surely we can adjust these to fit the memory resize without losing capability?

Reply:
you can correct the values if needed by setting zfs_dirty_data_max[_max] and it will take effect in arc_tuning_update fn

if ((zfs_dirty_data_max_max) &&
+	    (zfs_dirty_data_max_max != dirty_data_max_max))
+		dirty_data_max_max = zfs_dirty_data_max_max;
+
+	if ((zfs_dirty_data_max) &&
+	    (zfs_dirty_data_max != dirty_data_max))
+		dirty_data_max = zfs_dirty_data_max;